src/emscatter.jl

```diff
@@ -0,0 +1,296 @@
+# In this tutorial, we will learn:
+# 
+#   * How to formulate the weak form for time-harmonic electromagnetic problem
+#   * How to implement perfectly matched layers (PML)
+#   * How does periodic boundary condition work in Gridap
+# 
+# # Problem statement
+# 
+# The problem that we are trying to solve is the plane wave (TM-polarized $H_{inc}$) scattering of a cylinder dielectric (of radius $R$ and permittivity $\varepsilon$), as illustrated below. The computational cell is of height $H$ and length $L$, and we designate a perfectly matched layer (PML) thickness of $d_{pml}$.
+# 
+# ![](../assets/emscatter/Illustration.png)
+# 
+# The governing equation of this scaler electromagnetic problem in 2D is the Helmholtz equation: 
+#
+# ```math
+# \left[-\nabla\cdot\frac{1}{\varepsilon(x)}\cdot\nabla -k^2\mu(x)\right] H = f(x),
+# ```
+#
+# where $k=\omega/c$ is the wave number in free space and $f(x)$ is the source term
+# 
+# The boundary conditions are zero Dirichlet boundary on the top and bottom side $\Gamma_D$ and periodic boundary condition on the left ($\Gamma_L$) and right side ($\Gamma_R$). The PML is implemented by a coordinate streching: 
+#
+# ```math
+# \frac{\partial}{\partial x}\rightarrow \frac{1}{1+\mathrm{i}\sigma(u)/\omega}\frac{\partial}{\partial x},
+# ```
+#
+# ```math
+# \frac{\partial}{\partial y}\rightarrow \frac{1}{1+\mathrm{i}\sigma(u)/\omega}\frac{\partial}{\partial y},
+# ```
+#
+# where $u$ is the depth into the PML, $\sigma$ is a profile function (here we chose $\sigma(u)=\sigma_0u^2$) and different derivative denotes different absorption direction. 
```

Comment:
At a finite mesh resolution, PML reflects some waves, and the standard technique to mitigate this is to "turn on" the PML absorption gradually—in this case we use a quadratic profile.  The amplitude σ_0 is chosen so that in the limit of infinite resolution the "round-trip" reflections are exp(-2 σ_0 L) ≈ ....

Reply:
I usually nondimensionalize `u` here to be `x/L` where `L` is the thickness.